binds/python/bind_immutable.cpp

```diff
@@ -143,7 +144,32 @@ void bind_immutable_module(py::module& m) {
             "can be passed:\n"
             "- morphio.IterType.depth_first (default)\n"
             "- morphio.IterType.breadth_first (default)\n"
-            "iter_type"_a = IterType::DEPTH_FIRST);
+            "iter_type"_a = IterType::DEPTH_FIRST)
+
+        // Make class picklable
+        .def(py::pickle(
+            [](const morphio::Morphology& morphology) {  // __getstate__
+                // Return a tuple that fully encodes the state of the object
+                if (!morphio::mut::writer::_shouldBeWritten(morphology))
+                    return py::make_tuple("");
+                std::ostringstream oss;
+                morphio::mut::writer::_asc_to_stream(morphology, oss);
```

Comment:
I'm concerned we're using ASC for the serialization, since it can't capture what `SWC` encodes, which means that we potentially run problems.

Also, how does this handle `mitochondria` and `endoplasmicReticulum`?

Reply:
It does not handle stuff beyond xyzr. If we want to do this we need to use another approach.
I've opted for using ASC as the serialization medium for its simplicity but maybe I should already start digging further.

Reply:
> [...] Indeed, we've recently added some multiprocessing features to NeuroM ([BlueBrain/NeuroM@4f01b4c]

Pickling/serialization is always hard.  I"m wondering if it would be possible to change the multiprocessing to not need pickling - only pass around filenames?

This PR could then be put off, since I think it will become more complicated than its current form.